src/Microsoft.VisualStudio.Services.Agent/JobServerQueue.cs

```diff
@@ -186,16 +186,16 @@ public void QueueWebConsoleLine(string line)
             {
                 Trace.Verbose("Enqueue web console line queue: {0}", line);
                 _webConsoleLineQueue.Enqueue(line);
-            }
 
-            // Too many web console lines enqueued.
-            // Signal dequeue task to run immediately in case of the task is still waiting for deplay
-            if (_webConsoleLineQueue.Count >= _webConsoleLineQueueForceProcessThreshold)
-            {
-                Trace.Verbose("Web console line queue has {0} lines enqueued, reach force process threshold {1}, signal dequeue task to process them right now.", _webConsoleLineQueue.Count, _webConsoleLineQueueForceProcessThreshold);
-                if (_webConsoleLineQueueSemaphore.CurrentCount == 0)
+                // Too many web console lines enqueued.
+                // Signal dequeue task to run immediately in case of the task is still waiting for deplay
+                if (_webConsoleLineQueue.Count >= _webConsoleLineQueueForceProcessThreshold)
```

Comment:
I wish there was a way to avoid calling Release under the lock. A reader writer lock might work around it. There's some optimization to be made here I just don't know what it is offhand. Looks good for now.